jax_layers/nanogpt/model.py

```diff
@@ -0,0 +1,111 @@
+"""NanoGPT model implementation using JAX and Flax."""
+
+from typing import Optional, Tuple
+
+import jax
+import jax.numpy as jnp
+from flax import linen as nn
```

Comment:
In this library, we're specifically targetting the newer `NNX` API, not the older `linen` API.